src/mlpack/methods/ann/loss_functions/hinge_loss.hpp

```diff
@@ -0,0 +1,102 @@
+/**
+ * @file methods/ann/loss_functions/hinge_loss.hpp
+ * @author Anush Kini
+ *
+ * Definition of the Hinge Loss Function.
+ *
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license.  You should have received a copy of the
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+
+#ifndef MLPACK_METHODS_ANN_LOSS_FUNCTION_HINGE_LOSS_HPP
+#define MLPACK_METHODS_ANN_LOSS_FUNCTION_HINGE_LOSS_HPP
+
+#include<mlpack/prereqs.hpp>
+
+namespace mlpack {
+namespace ann /** Artificial Neural Network. */ {
+
+/**
+ * Computes the hinge loss between y_true and y_pred. Expects y_true to be
+ * either -1 or 1. If y_true is either 0 or 1, a temporary conversion is made to
+ * calculate the loss.
```

Comment:
Do you think you can use doxygen math `$$` around `y_true` and `y_pred` also maybe it would be useful to write the complete formula here. Could make it easier to understand what `y_true` and `y_pred` is and also how they fit into everything.

Reply:
That's a great idea. I know about using ```\f$``` to type formulas in doxygen. Is this the same as ```$$```?
If not, could you point me to an example of this in the mlpack codebase?

Reply:
Right, either is fine you can just go with`\f$`.